hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DebugAdmin.java

```diff
@@ -306,7 +306,7 @@ int run(List<String> args) throws IOException {
             new BufferedOutputStream(Files.newOutputStream(srcMeta.toPath()),
                 smallBufferSize));
         BlockMetadataHeader.writeHeader(metaOut, checksum);
-        metaOut.close();
+        metaOut.flush();
```

Comment:
I think close itself is ok, we can avoid closing in the finally block if the stream is already closing, rather than flushing here & then closing below, close would be calling flush before closing itself.

in general I don't think it isn't a big problem, maybe just dereference the metaOut after close
```
        metaOut.close();
        metaOut = null;
```

Reply:
Good idea，I had change it @ayushtkn

Reply:
I think you got me wrong, only 1 line was required, finally should have remained, if there is an exception before close, the stream would remain open otherwise

Reply:
Yes，I misstaked，keep finally block and add 1 line. @ayushtkn